fabric-network/src/impl/query/singlequeryhandler.js

```diff
@@ -26,39 +25,30 @@ class SingleQueryHandler {
 
 		const startPeerIndex = this._currentPeerIndex;
 		const errorMessages = [];
-		const options = {requestTimeout: 3000};
-		// use the timeout as the requestTimeout or let default
-		if (Number.isInteger(this._options.timeout)) {
-			options.requestTimeout = this._options.timeout * 1000; // in ms;
-		}
 
 		for (let i = 0; i < this._peers.length; i++) {
 			const peerIndex = (startPeerIndex + i) % this._peers.length;
 			this._currentPeerIndex = peerIndex;
-			const peer = this._peers[peerIndex];
-
-			logger.debug('%s - query sending to peer %s', method, peer.name);
-			const results = await query.send({targets:[peer]}, options);
-
-			if (results.errors.length > 0) {
-				logger.error('%s - problem with query to peer %s error:%s', method, peer.name, results.errors[0]);
-				// since only one peer, only one error
-				errorMessages.push(results.errors[0].message);
-				continue;
-			}
-
-			const endorsementResponse = results.responses[0];
 
-			if (!endorsementResponse.endorsement) {
-				logger.debug('%s - peer response status: %s message: %s',
-					method,
-					endorsementResponse.response.status,
-					endorsementResponse.response.message);
-				throw new Error(endorsementResponse.response.message);
+			const peer = this._peers[peerIndex];
+			logger.debug('%s - sending to peer %s', method, peer.name);
+
+			const results = await query.evaluate([peer]);
+			const result = results[peer.name];
+			if (result instanceof Error) {
+				errorMessages.push(result.toString());
+			} else {
+				if (result.status < 400) {
+					logger.debug('%s - return peer response status: %s', method, result.status);
+					return result.payload;
+				} else if (result.status < 500) {
+					logger.debug('%s - throw peer response status: %s message: %s', method, result.status, result.message);
+					throw Error(result.message);
+				} else {
+					logger.debug('%s - peer response message: %s', method, result.message);
+					errorMessages.push(result.message);
```

Comment:
Still not behaving the same as previously. See comment in `roundrobinqueryhandler.js`